### PR TITLE
Add table of contents to upgrade guides

### PR DESCRIPTION
**Pull request by smithandre**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | NA
| License       | MIT
| Doc PR        | NA

Added a table of contents to the upgrade guides, which can help us find relevant upgrade information a bit faster. It's very basic and straight forward, as I don't want to add too much overhead in terms of maintaining these guides.

### CODE REVIEW

File: UPGRADE-4.1.md [hunk 1]
```diff
@@ -1,5 +1,20 @@
 UPGRADE FROM 4.0 to 4.1
 =======================
+# Table of Contents
```
Comment (nicolas-grekas) [inline UPGRADE-4.1.md]: missing blank line before (same in other files)

**Review by nicolas-grekas — COMMENTED** [review]
I'm not sure we should merge this: it looks nice, but it looks hard to maintaint.

**Comment by ro0NL** [comment]
there's also https://github.com/symfony/symfony/issues/32840, i'd prefer a generated ToC as well (to reduce maintenance load)

once tooling/infrastructure is in place, we could leverage e.g. https://github.com/ekalinin/github-markdown-toc

**Comment by nicolas-grekas** [comment]
Closing as explained, thanks for proposing.

**Pull request closed without merging**